packages/solutions/module-tools/src/builder/clear.ts

```diff
@@ -8,15 +8,29 @@ export const clearDtsTemp = async () => {
 
 export const clearBuildConfigPaths = async (
   configs: BaseBuildConfig[],
-  noClear: boolean,
+  options?: {
+    noClear?: boolean;
+    projectAbsRootPath?: string;
+  },
 ) => {
+  const { noClear = false, projectAbsRootPath = process.cwd() } = options ?? {};
   const { fs } = await import('@modern-js/utils');
 
   if (noClear) {
     return;
   }
 
   for (const config of configs) {
-    await fs.remove(config.outDir);
+    if (projectAbsRootPath === config.outDir) {
```

Comment:
Can `config.outDir` be a relative path?

Reply:
Module tools will convert all paths to absolute paths